src/python/dxpy/scripts/dx.py

```diff
@@ -2514,7 +2514,13 @@ def install(args):
 def uninstall(args):
     app_desc = get_app_from_path(args.app)
     if app_desc is None:
-        parser.exit(1, 'Could not find the app\n')
+        #in case we lost permission for this app, try to remove it by Id.
+        try:
+            dxpy.api.app_uninstall(args.app)
+            print('Uninstalled application with id: ' + args.app)
+        except Exception as error:
+            parser.exit(1, 'Could not find the app! If application still in list, '
+                        'please try uninstall it providing appropriate Id\n' + str(error))
```

Comment:
Please consider rephrasing this as:
"Could not find the app.\n If the app still appears on the list, please try to uninstall it by providing the appropriate Id" + str(error)

However, if the user does not have access to the app, would he still be able to obtain the app id with `dx describe appname`? If not, then we might need a different message.